src/settings/LoanHistory/LoanHistoryForm.js

```diff
@@ -69,57 +76,43 @@ class LoanHistoryForm extends Component {
     );
   }
 
-  renderPeriod() {
+  toggleCheckbox = () => {
+    this.setState(({ checked }) => ({
+      // eslint-disable-next-line react/no-unused-state
+      checked: !checked
+    }));
+  }
+
+  getCurrentValues() {
+    const { store } = this.props.stripes;
+    const state = store.getState();
+    return getFormValues('loanHistoryForm')(state) || {};
+  }
+
+  renderPeriod(name) {
     return (
       <div
-        data-test-loans-section-loan-period
+        data-test-period-section
         className={css.periodContainer}
       >
         <Period
-          inputValuePath="intervalValue"
-          selectValuePath="intervalType"
+          inputValuePath={`${name}.duration`}
+          selectValuePath={`${name}.intervalId`}
           intervalPeriods={this.generateOptions(this.selectedPeriods, 'ui-circulation.settings.loanHistory.selectInterval')}
           inputSize={4}
-          selectSize={6}
+          selectSize={7}
         />
+        <div><FormattedMessage id="ui-circulation.settings.loanHistory.after" values={{ name }} /></div>
       </div>
     );
   }
 
-  updateClosingTypes = types => {
-    const updatedType = { label: this.renderPeriod(), value: closingTypesMap.INTERVAL };
-
-    return types.map(type => (type.value === closingTypesMap.INTERVAL ? updatedType : type));
-  };
-
-  renderClosingTypes = ({ meta }) => {
-    const updatedClosingTypes = this.updateClosingTypes(closingTypes);
-    const items = updatedClosingTypes.map((type, index) => (
-      <Row key={`row-${index}`}>
-        <Col xs={12}>
-          <Field
-            component={RadioButton}
-            label={<FormattedMessage id={type.label} />}
-            name="closingType"
-            type="radio"
-            id={`${type.value}-radio-button`}
-            value={type.value}
-            labelClass={css.closingTypeField}
-          />
-        </Col>
-      </Row>
-    ));
-
-    return (
-      <div>
-        <FormattedMessage
-          tagName="p"
-          id="ui-circulation.settings.loanHistory.anonymize"
-        />
-        {items}
-        {meta.error && <div className={css.error}>{meta.error}</div>}
-      </div>
-    );
+  updateClosingTypes(name) {
```

Comment:
`Update` part seems wrong for me for the naming as you effectively use it for getting some types out. Name it in another way.